working/augmentation-libraries/feature-specification.md

```diff
@@ -321,82 +322,150 @@ doesn't, it must repeat the type from the augmented definition__.
 
 ### Augmented Expression
 
-The exact result of an `augmented` expression depends on what is being
-augmented, but it generally follows the same rules as any normal identifier:
-
-*   **Augmenting getters**: Within an augmenting getter `augmented` invokes
-    the augmented getter and evaluates to its return value. If augmenting a
-    variable with a getter, this will invoke the implicitly induced getter
-    from the augmented variable declaration.
-
-*   **Augmenting setters**: Within an augmenting setter `augmented` must be
-    followed by an `=` and will directly invoke the augmented setter. If
-    augmenting a variable with a setter, this will invoke the implicitly
-    induced setter from the augmented variable declaration.
-
-*   **Augmenting fields**: Within an augmenting variable declaration,
-    `augmented` can only be used in an initializer expression, and refers
-    to the augmented variable's initializing expression, which is
-    immediately evaluated.
-
-    It is a compile-time error to use `augmented` in an augmenting
-    variable's initializer if the member being augmented is not a variable
-    declaration with an initializing expression.
+The identifier `augmented` is a reserved word, with associated special syntax,
+when it occurs inside an expression or body of an augmenting variable, getter,
+setter, function, operator or constructor declaration.
```

Comment:
Just to be clear, the "or body" means that `augmented` is *not* a reserved word inside the *signature* of a declaration, right? So you could have, for example:

```dart
void foo(int x) {}

augment void foo(int augmented) {}
```

Do we want to allow parameters or type annotations to use `augmented` in augmenting declarations? I can't see any good use for it.

Reply:
Yes, that is the goal. The `augmented` word is only reserved *inside the body*, and in any expressions outside of the body (default value expression, initializer list/redirection expressions for generative constructors), but not in the signature of the declaration.

I don't necessarily see a use for `augmented` in the parameter list, but I also don't see a use for reserving it outside of where it can be used.
The words `await` and `yield` are only reserved inside the asynchronous body, they're not reserved in the parameter list. 
They are also reseved in the signatures of nested functions, but not those function's body. That might be a mistake, but it avoids you declaring a `void await(Object? _){}` function inside an `async` function.
It also avoids you declaring a `void foo({bool await = false}) {...}` function, even if you need one, you need to do `var foo = (() => ({bool await = false}) {...})();` instead, which you can.
(I'd like to allow nested function declarations to not treat the `augmented`, `yield` and `await` words as reserved in their signatures, even if they can't use it as a name, so that it's possible to declare a function with `augmented`/`await` as a named paramete name. Since you can declare them elsewhere, you might *need* to create a function like that, and the only workaround is to have a function declared outside of the reserved word scope that renames the parameter for you. But that's a different and bigger issue, and for `augmented` it could mean that you can *shadow* the `augmented` of the outer function with a parameter. Introducing `augmented` as a reserved word in entire expressions and bodies is probably the simplest, and therefore currently best, choice.)

I am reserveing `augmented` in initializer list and default value expressions because it would be possible to allow augmenting those and referring to the original expression. Again it's maybe not useful, but I want to keep the door open, and "reserved in any runtime code" (things that get executed or evaluated) is easier to remember than reserving it in initialier list initializer expressions, but not assert expressions.

So "body and expressions" covers that and is fairly straight line for users to remember.

(We _could_ choose to allow conditionally reserved words as parameter names, even in scopes where they are reserved, so you could call a function `foo(await: true, yield: 42)` even inside an `async` function. Any name you *can* use as a parameter name anywhere can also be used as a named argument name anywhere. Or even as a member name, `object.await(true).yield(42)` inside an async function, the prior `.` or following `:` making it be treated as a member-name, not a reserved word. Globally reserved words cannot be used that way. Or maybe they can. If we allow *escaping* reserved words into identifiers, like `this.\for(\await: true, \yield: 42, \if: isTuesday)`, so we  can use it for interop with languages that have different reserved words, then allowing *unescaped* named parameters and member access with reserved words makes the interop experience nicer. Shucks to be a syntax highlighter, but if it can latch onto the `.word` and `word:` patterns, it just might work!)